SM Test Fixes

labkey-danield opened the pull request:
#### Rationale
Fixing some tesst issue

#### Related Pull Requests
- https://github.com/LabKey/limsModules/pull/1688

#### Changes
- Assert that QueryGrid.clearAllSelections does what it says it will do.
- Per suggestions from Allan use Locator.byClass rather than Locator.tagWitClass.
- SampleTypeAPIHelper.getRowIdsForSamples use json to pass the list of sample ids to the API
- In TestDataGenerator.isNameInvalidLocal change regular expression to work if " -" is the last character in the domain name.

labkey-danield committed:
Merge branch 'develop' into fb_smTestFixes

labkey-danield committed:
Merge branch 'develop' into fb_smTestFixes

labkey-danield committed:
Merge branch 'develop' into fb_smTestFixes

labkey-danield committed:
Use json when calling the API.

labkey-danield committed:
Catch ' -' at the end of the domain name as an invalid name.

labkey-danield committed:
Merge branch 'develop' into fb_smTestFixes

labkey-danield committed:
More logging.
Simpler selectors.

labkey-danield committed:
Assert that QueryGrid.clearAllSelections works.
Use a valid domain name in SMSampleTypeDesignerTest.testParentAndSourceAliasOptions
Change the order of clearing the filters and selections in SMProMoveAssayRunsTest.configureGrid

labkey-danield committed:
Merge branch 'develop' into fb_smTestFixes

XingY commented on src/org/labkey/test/util/exp/SampleTypeAPIHelper.java:
```diff
@@ -133,10 +134,29 @@ public static Map<Integer, String> getSortedRowIdToSamplesMap(String containerPa
+        while (iterator.hasNext()) {
+            String sampleName = iterator.next();
+
+            json.append("\"");
```
Could use `EscapeUtil.toJSONStr`

labkey-danield replied:
Done.

labkey-danield committed:
Implementing code review feedback.
Dismissing banner messages after moving assay runs.
Using EscapeUtil.toJSONStr to build json string.
Added BannerHelper.dismissAllBannerMessages
